NDTensors/ext/NDTensorMetal/array.jl

```diff
@@ -0,0 +1,41 @@
+using NDTensors
+using LinearAlgebra
+
+function LinearAlgebra.norm(x::Metal.MtlArray)
+  v = real(eltype(x))(0.0)
+  for i in x
+    v += real(conj(i) * i)
+  end
+  return sqrt(v)
+end
+
+function transform(old::Metal.MtlArray, new::Metal.MtlArray, func::Function)
+  @assert length(old) == length(new)
+  for i in 1:length(old)
+    new[i] = func(old[i])
+  end
+  return new
+end
+
+function Base.real(x::Metal.MtlArray)
```

Comment:
This is already defined:
```julia
julia> A = mtl(randn(ComplexF32, 2, 2))
2×2 MtlMatrix{ComplexF32}:
 -0.420252+0.722033im   1.17723+1.01282im
 -0.227524+0.253891im  -0.49268+2.55224im

julia> real(A)
2×2 MtlMatrix{Float32}:
 -0.420252   1.17723
 -0.227524  -0.49268
```
(I would guess the definitions below this one are also already defined for `MtlArray`).